recipes/soplex/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "BOOST": [True, False],
+        "GMP": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "BOOST": False,
+        "GMP": False
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "clang": "4",
+            "apple-clang": "7",
+        }
+
+    def _determine_lib_name(self):
+        if self.options.shared:
+            return "soplexshared"
+        elif self.options.get_safe("fPIC"):
+            return "soplex-pic"
+        else:
+            return "soplex"
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        replace_in_file(self, "CMakeLists.txt", "GMP_INCLUDE_DIRS", "gmp_INCLUDE_DIRS")
+        replace_in_file(self, "CMakeLists.txt", "GMP_LIBRARIES", "gmp_LIBRARIES")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("zlib/1.2.13", transitive_headers=True)
+        if self.options.GMP:
+            self.requires("gmp/6.2.1")
+        if self.options.BOOST:
+            self.requires("boost/1.81.0")  # also update Boost_VERSION_MACRO below!
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["GMP"] = self.options.GMP
+        tc.variables["BOOST"] = self.options.BOOST
+        tc.variables["Boost_VERSION_MACRO"] = "108100"
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build(target=f"lib{self._determine_lib_name()}")
+
+    def package(self):
+        copy(self, pattern="LICENSE", src=self.source_folder, dst=join(self.package_folder, "licenses"))
+        copy(self, pattern="soplex.h", src=join(self.source_folder, "src"), dst=join(self.package_folder, "include"))
```

Comment:
Would it not make sense to use the CMake install that is provided upstream? https://github.com/scipopt/soplex/blob/4ab99fc9f52fcbcc738a90e59cbba1862eb71d92/src/CMakeLists.txt#L215 for example?

Reply:
This CMake install requires that all libs (shared/nonshared/fpic/...) are built and installs all of them

Reply:
There should be an install component arg you can pass https://docs.conan.io/2/reference/tools/cmake/cmake.html?highlight=cmake#conan.tools.cmake.cmake.CMake.install

Reply:
I do not understand how components are helping here. The upstream CMake does not use components. There is no `COMPONENT` keyword in any of the `install` commands. My understanding of components is based on this: https://discourse.cmake.org/t/understanding-the-cmake-component-keyword-in-the-install-command/971

Reply:
Conan tries to use uniform language for everything, this command the "components" is equivalent to "target" 

This is explain in the docs about how the conan version is translated to actual cmake :)

Reply:
I looked into all the generated `cmake_install.cmake`. What is in common is

```
if(CMAKE_INSTALL_COMPONENT STREQUAL "Unspecified" OR NOT CMAKE_INSTALL_COMPONENT)
  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/include/soplex" TYPE FILE FILES
    "/Users/ivhedtke/.conan/data/soplex/6.0.3/_/_/build/aeb351eac3a1f007eb42261653c05a7909939cfb/src/src/soplex/array.h"
    "/Users/ivhedtke/.conan/data/soplex/6.0.3/_/_/build/aeb351eac3a1f007eb42261653c05a7909939cfb/src/src/soplex/basevectors.h"
```

```
if(CMAKE_INSTALL_COMPONENT STREQUAL "Unspecified" OR NOT CMAKE_INSTALL_COMPONENT)
  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/lib" TYPE STATIC_LIBRARY FILES "/Users/ivhedtke/.conan/data/soplex/6.0.3/_/_/build/aeb351eac3a1f007eb42261653c05a7909939cfb/build/Release/lib/libsoplex.a")
```

```
if(CMAKE_INSTALL_COMPONENT STREQUAL "Unspecified" OR NOT CMAKE_INSTALL_COMPONENT)
  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/lib" TYPE STATIC_LIBRARY FILES "/Users/ivhedtke/.conan/data/soplex/6.0.3/_/_/build/aeb351eac3a1f007eb42261653c05a7909939cfb/build/Release/lib/libsoplex-pic.a")
```

there only one possible value (from hat I see in the generated files) for `component` is "Unspecified" which is the same as omitting this argument. Not specifying this argument or setting it to "Unspecified" results in a failed install as the install-process tries to install cmd tools (bin/soplex) and all versions (fPIC/shared) of all libs. Setting it to anything else, like `libsoplex`, `soplex` or the `-pic` variants, installs nothing as they are not valid components.

I really have no idea how to use the `component` argument. I read the docu multiple times and tried lots of arguments.